yt/yt/server/query_tracker/handler_base.cpp

```diff
@@ -79,33 +79,54 @@ TSharedRef TruncateRowset(TSharedRange<NTableClient::TUnversionedRow>& rows)
     return TSharedRef::MakeEmpty();
 }
 
-void ProcessRowset(TFinishedQueryResultPartial& newRecord, TWireRowset wireSchemaAndSchemafulRowset)
+void ProcessRowset(TFinishedQueryResultPartial& newRecord, TWireRowset wireSchemaAndSchemafulRowset, i64 ValueLengthLimit)
 {
-    // Process schema.
-    auto reader = CreateWireProtocolReader(wireSchemaAndSchemafulRowset.Rowset);
-    auto schema = reader->ReadTableSchema();
-    auto schemaNode = ConvertToNode(schema);
-    // Values in tables cannot have top-level attributes, but we do not need them anyway.
-    schemaNode->MutableAttributes()->Clear();
-    newRecord.Schema = ConvertToYsonString(schemaNode);
-
-    // Process schemaful rowset.
-    auto rowset = reader->Slice(reader->GetCurrent(), reader->GetEnd());
-    auto schemaData = IWireProtocolReader::GetSchemaData(schema);
-    auto rows = reader->ReadSchemafulRowset(schemaData, /*captureValues*/ false);
-    TDataStatistics dataStatistics;
-    dataStatistics.set_row_count(rows.size());
-    dataStatistics.set_data_weight(GetDataWeight(rows));
-    newRecord.DataStatistics = ConvertToYsonString(dataStatistics);
-    newRecord.IsTruncated = wireSchemaAndSchemafulRowset.IsTruncated;
-    if (rowset.Size() <= MaxStringValueLength) {
-        // Fast path. Copy full rowset.
-        newRecord.Rowset = TString(rowset.ToStringBuf());
-    } else {
-        // Slow path. Truncate rowset.
+    try {
+        YT_LOG_DEBUG("Processing wire rowset");
+        // Process schema.
+        YT_LOG_DEBUG("Reading schema");
+        TWireProtocolOptions readerOptions;
+        readerOptions.MaxStringValueLength = ValueLengthLimit;
```

Comment:
This is still a limit only on QT side. I am still very concerned that you can run a query with a huge result that will be piped from YQL agent to QT, only to produce an error. 

Let's at least test (locally and/or on testing clusters) a few things:
1. Is it possible to produce outputs larger than 128M? Maybe there will be another limit somewhere based on the max row size for both static and dynamic tables.
2. What happens when you run a query with a huge output, like 10G, how do the other queries fare?

If you and @Krisha11 are collectively okay with this behavior, I am fine with merging this, but I think it is dangerous.

Reply:
>This is still a limit only on QT side. I am still very concerned that you can run a query with a huge result that will be piped from YQL agent to QT, only to produce an error.

Assume that there is a separate limit on each engine side. Engine-specific

>Is it possible to produce outputs larger than 128M? Maybe there will be another limit somewhere based on the max row size for both static and dynamic tables.

It is. YQL team said that they have some limit like 2GB (For a single value). I am not sure if it applies to us

>What happens when you run a query with a huge output, like 10G, how do the other queries fare?

It is a separate issue that should be tested and resolved separately. Let's assume that Query Tracker dies in an unexpected way that requires investigation and solution

Huge output is possible already. Example is a table with 8 string columns, each of 16 mb, with 10000 rows. It produces 1.28TB of output. YQL query `SELECT *` will face no issues in the code in this PR

>I am fine with merging this, but I think it is dangerous.

Can you elaborate? This PR doesn't change amount of data transferred between YQL Agents and Query Trackers. It fixes validation error and freezing a query in an "active" state

Reply:
You are right that there are other ways to transfer huge amounts of data, that's exactly my point — we should forbid it. But ok, you are probably right that it is best to do it in a separate PR.

Reply:
Tested `select *` on a table with 100 16mb rows (1.56gb). YQL agent failed with OOM. Query failed with `Socket read error. Connection reset by peer`